common/achievement/metrics.ts

```diff
@@ -0,0 +1,28 @@
+import { Metric } from './types';
+
+export const metrics: Map<string, Metric>[] = [];
```

Comment:
Why Map Array?

Reply:
I think it rather makes sense to have

metricByName: Map<string, Metric>
metricsByAction: Map<ActionID, Metric[]>